tableauserverclient/server/endpoint/endpoint.py

```diff
@@ -39,6 +39,7 @@ def _make_request(self, method, url, content=None, request_object=None,
             parameters['data'] = content
 
         server_response = method(url, **parameters)
+        self.parent_srv._namespace.detect(server_response.content)
```

Comment:
I see, the first request, of any kind, checks and caches itself, then we shortcut on subsequent requests.

I like the deisgn.

Reply:
yup.  I think the only place where we might get in trouble is if the first request isn't the use server version workflow.  At this point, I want to make that the default behavior. @LGraber any objections to always determining the server version to help solve this issue?

Reply:
Why would it be a problem if that wasn't the first call? I am not sure I see how this only happens on the first call?